src/Axe/Views/Compiler/Solve.cshtml

```diff
@@ -1,87 +1,102 @@
 @using Axe.Models;
 @model Axe.ViewModels.CompilerVm.CodeBlockSolveVm
+
 @{
     ViewData["Title"] = "Solve task #" + Model.Id;
 }
 
 @section Scripts
 {
     <script src="~/js/compiler.solve.js"></script>
+    @if (Model.isMonitoringMode && ViewData["observedUserId"] != null)
+    {
+        <script src="~/js/compiler.solve.observer.js"></script>
+    }
+    else
+    {
+        <script src="~/js/compiler.solve.user.js"></script>
+    }
+}
+
+@if (Model.isMonitoringMode && ViewData["observedUserId"] != null)
+{
+    <input id="observedUserId" value="@ViewData["observedUserId"]" hidden="hidden" />
 }
 
+<input id="taskId" asp-for="Id" value="@Model.Id" hidden />
+<input asp-for="Task" value="@Model.Task" hidden />
+
 <div class="row" style="margin-top: 1em;">
-    <div class="col-md-12">
-        <div class="panel panel-default">
-            <div class="panel-heading"><b>Task:</b> @Model.Id</div>
-            <div class="panel-body">@Model.Task</div>
+    <div class="col-md-4">
+        <div class="col-md-12">
+            <div class="panel panel-default">
+                <div class="panel-heading"><b>Task #: </b>@Model.Id</div>
+                <div class="panel-body">@Model.Task</div>
+            </div>
         </div>
-    </div>
-</div>
 
-<form id="formSolve" asp-controller="Compiler" asp-action="Solve" method="post">
-    <input asp-for="Id" value="@Model.Id" hidden />
-    <input asp-for="Task" value="@Model.Task" hidden />
-    
-    <div class="row">
-        <div class="form-group col-md-12">
-            <label asp-for="Technology" class="col-md-3"></label>
+        <div class="form-group">
+            <label asp-for="Technology" class="col-md-12"></label>
             <input asp-for="SelectedTechnologyId" value="@Model.Technology.Id" type="hidden" />
-            <div class="col-md-9">
+            <div class="col-md-12">
                 <input id="selectedTechnology" value="@Model.Technology.Name" disabled="disabled" class="form-control disabled" />
             </div>
         </div>
     </div>
-    
-    <div class="row">
-        <div class="form-group col-md-12">
-            <label asp-for="SourceCode" class="col-md-12"></label>
-            <div class="col-md-12">
-                <textarea id="sourceCodeTextArea" asp-for="SourceCode" rows="30" class="form-control">@Model.SourceCode</textarea>
+
+    <div class="col-md-8">
+        <div class="form-group">
+            <label asp-for="SourceCode"></label>
+            <div>
+                <textarea id="sourceCodeTextArea" asp-for="SourceCode" rows="15" class="form-control">@Model.SourceCode</textarea>
             </div>
```

Comment:
area for code is way too small. use entire row and i don't think 15 lines in enough for any non trivial code -dotnetfiddle has 30, collabedit has 48!